validate.py

```diff
@@ -187,28 +212,27 @@ def check_items(self, report_dir=None):
                 #: Initialize empty dictionary for this item
                 self.report_dict[itemid] = {}
 
-                checker = checks.ItemChecker(item, self.metatable_dict, credentials.DB)
+                checker = checks.ItemChecker(item, self.metatable_dict)
+                retry(lambda: checker.setup(credentials.DB))
+
+                #: TODO: add each method and it's args to a list, then iterate through the list (DRY)
 
                 #: Run the checks on this item
-                checker.tags_check(self.tags_to_delete, self.uppercased_tags, self.articles)
-                checker.title_check()
-                checker.folder_check(self.itemid_and_folder)
-                checker.groups_check()
-                checker.downloads_check()
-                checker.delete_protection_check()
-                checker.metadata_check()
-                checker.description_note_check(self.static_note, self.shelved_note)
+                retry(lambda: checker.tags_check(self.tags_to_delete, self.uppercased_tags, self.articles))
```

Comment:
If this line raises an exception (after retrying three times) then 222-231 wouldn't run, correct? It would skip down to the finally code. I'm not sure if this is intended or not. The only way that I can think of to get around this is to not raise an error if retry fails.

Reply:
I'm not saying that you should change the code, just an observation.

Reply:
Yeah, that was my understanding of how the error handling would work. 

I'm assuming that if the network doesn't come back up after 14 seconds, it's probably well and truly dead and so the program should probably just completely bomb out. I agree with your work around statement.

Reply:
Mainly I had it fail twice yesterday about halfway through the fix side of things and decided it was time time to do something about it :)